ebean-test/src/test/java/org/tests/batchinsert/TestBatchInsertFlush.java

```diff
@@ -359,4 +369,79 @@ public void testFlushOnSetProperty() {
       txn.end();
     }
   }
+
+  @Test
+  public void testFlushWithOrder() {
+
+    ResetBasicData.reset();
+
+    try (Transaction txn = DB.beginTransaction(TxScope.requiresNew())) {
+      // do anything, that enables batch mode
+      // in ebean 14
+      DB.insertAll(List.of(new Acl("test")));
+      // does the same as txn.setBatchMode(true);
+
+      // Save contact without group assigned
+      Contact contact = new Contact();
+      contact.setCustomer(DB.find(Customer.class).setMaxRows(1).findOne());
+      DB.save(contact);
+
+      // create new group with fixed id
+      ContactGroup group = new ContactGroup();
+      group.setName("Max");
+      group.setId(42);
+      DB.save(group);
+
+      // assign new group to contact
+      contact.setGroup(group);
+      DB.save(contact);
+      // Ebean holds in it's internal batch queue Contact:0. ContactGroup:1
+
+      txn.commit();
+    }
+  }
+
+
+  @Test
+  public void testWithBatch() {
+
+    ResetBasicData.reset();
+
+    Map<Integer, String> contactCsv = Map.of(123, "Noemi", 124, "Roland");
+    Map<Integer, String> contactGroupCsv = Map.of(42, "Praml", 43, "Max");
+    Map<Integer, Integer> mappingCsv = Map.of(123, 42);
+    try (Transaction txn = DB.beginTransaction(TxScope.requiresNew())) {
+      txn.setBatchMode(true);
+
+      Customer someCustomer = DB.find(Customer.class).setMaxRows(1).findOne(); // just to get no error
+      // Save contact without group assigned
+
+      Map<Integer, Contact> createdContacts = new HashMap<>();
+      contactCsv.forEach((id, name) -> {
+        Contact contact = new Contact();
+        contact.setFirstName(name);
+        contact.setId(id);
+        contact.setCustomer(someCustomer); // customer is mandatory
+        DB.save(contact);
```

Comment:
Here we create the contacts and same them in a map for further mapping (I know, there are better ways, but this is what our code does. There are up to 100 objects in a temporary map)